x/projects/keeper/project.go

```diff
@@ -51,115 +62,204 @@ func (k Keeper) GetProjectForDeveloper(ctx sdk.Context, developerKey string, blo
 func (k Keeper) AddKeysToProject(ctx sdk.Context, projectID string, adminKey string, projectKeys []types.ProjectKey) error {
 	ctxBlock := uint64(ctx.BlockHeight())
 
-	nextEpoch, err := k.epochstorageKeeper.GetNextEpoch(ctx, ctxBlock)
+	project, _, err := k.getProjectForBlock(ctx, projectID, ctxBlock)
 	if err != nil {
-		return utils.LavaFormatError("AddKeysToProject: failed to get NextEpoch", err,
-			utils.Attribute{Key: "index", Value: projectID},
+		return utils.LavaFormatWarning("failed to add keys", err,
+			utils.Attribute{Key: "project", Value: projectID},
+			utils.Attribute{Key: "block", Value: ctxBlock},
 		)
 	}
 
-	// validate the admin key against the current project state; but make the
-	// changes on the future (= end of epoch) version if already exists. This
-	// ensures that we do not lose changes if there are e.g. multiple additions
-	// in the same epoch.
-
-	// validate admin key with current state
-	var project types.Project
-	if found := k.projectsFS.FindEntry(ctx, projectID, ctxBlock, &project); !found {
+	if !project.IsAdminKey(adminKey) {
 		return utils.LavaFormatWarning("failed to add keys",
-			fmt.Errorf("project not found"),
+			fmt.Errorf("requesting key must be admin key"),
 			utils.Attribute{Key: "project", Value: projectID},
 			utils.Attribute{Key: "block", Value: ctxBlock},
 		)
 	}
 
-	if !project.IsAdminKey(adminKey) {
-		return utils.LavaFormatWarning("failed to add keys",
-			fmt.Errorf("requesting key must be admin key"),
+	for _, projectKey := range projectKeys {
+		err := k.registerKey(ctx, projectKey, &project, ctxBlock)
+		if err != nil {
+			return utils.LavaFormatError("failed to register key to project", err,
+				utils.Attribute{Key: "project", Value: projectID},
+				utils.Attribute{Key: "block", Value: ctxBlock},
+			)
+		}
+	}
+
+	err = k.projectsFS.AppendEntry(ctx, projectID, ctxBlock, &project)
```

Comment:
we need to make sure the keys are registered in the beggining of the epoch, instead of append use modify entry to edit the project

Reply:
actually we can use append but make sure to use the block of the project and not ctxBlock